src/main/kotlin/no/skatteetaten/aurora/boober/service/DeployService.kt

```diff
@@ -165,12 +163,19 @@ class DeployService(
             val cmd = TagCommand("$dockerGroup/${it.artifactId}", it.version, it.releaseTo!!, dockerRegistry)
             dockerService.tag(cmd)
         }
-        val redeployResponse = redeployService.triggerRedeploy(deploymentSpec, openShiftResponses)
 
-        val redeploySuccess = if (redeployResponse.isEmpty()) true else redeployResponse.last().success
-        val totalSuccess = listOf(success, tagResult?.success, redeploySuccess).filterNotNull().all { it }
+        tagResult?.takeIf { !it.success }?.let {
+            return result.copy(tagResponse = it, reason = "Tag command failed")
+        }
+
+        val redeployResult = redeployService.triggerRedeploy(deploymentSpec, openShiftStatus)
+
+        if (!redeployResult.success) {
```

Comment:
What i do not like here it that openShiftStatus.openShiftReponses is right because we do mutability correct.